src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -981,17 +986,28 @@ def add_network_rule(cmd, client, resource_group_name, vault_name, ip_address=No
                 break
         if to_modify:
             rules.virtual_network_rules.append(VirtualNetworkRule(id=subnet))
+            to_update = True
 
     if ip_address:
         rules.ip_rules = rules.ip_rules or []
         # if the rule already exists, don't add again
-        to_modify = True
-        for x in rules.ip_rules:
-            if x.value == ip_address:
-                to_modify = False
-                break
-        if to_modify:
-            rules.ip_rules.append(IPRule(value=ip_address))
+        for ip in ip_address:
+            to_modify = True
+            for x in rules.ip_rules:
+                existing_ip_network = ip_network(x.value)
+                new_ip_network = ip_network(ip)
+                if new_ip_network.overlaps(existing_ip_network):
```

Comment:
Suddenly thinking of the relationship between `new_ip_network` and `existing_ip_network`. Should we prevent adding when `new_ip_network` overlap with `existing_ip_network` or `new_ip_network` is included by `existing_ip_network`?

Reply:
Verified in Portal. Portal uses overlapping check so I'm OK with this implementation